src/gax.ts

```diff
@@ -292,23 +342,148 @@ export class CallSettings {
   }
 }
 
+/**
+ * Validates passed retry options in preparation for eventual parameter deprecation
+ * converts retryRequestOptions to retryOptions
+ * then sets retryRequestOptions to null
+ *
+ * @param {CallOptions} options - a list of passed retry option
+ * @return {CallOptions} A new CallOptions object.
+ *
+ */
+
+export function checkRetryOptions(
+  options?: CallOptions,
+  gaxStreamingRetries?: boolean
+): CallOptions | undefined {
+  // options will be undefined if no CallOptions object is passed at call time
+  if (options) {
+    // if a user provided retry AND retryRequestOptions at call time, throw an error
+    if (gaxStreamingRetries) {
+      if (
+        options.retry !== undefined &&
+        options.retryRequestOptions !== undefined
+      ) {
+        throw new Error('Only one of retry or retryRequestOptions may be set');
+      } else {
+        if (options.retryRequestOptions !== undefined) {
+          // // Retry settings
+          if (options.retryRequestOptions.objectMode) {
+            console.log(
```

Comment:
For warnings, we don't use `console.log` since it's a library and excessive logging is not a good thing.  Please take a look at `src/warnings.ts` which has a way to emit each warning just once.

Reply:
Awesome. Will def make that change. I used `src/warnings.ts` in other places but for whatever reason didn't here. Is there a particular `warnType` you recommend using here, or a list of common `warnType`s we throw to choose from? Or is something like `UnsupportedWarning` appropriate to use?

Reply:
I think we don't need any type in this case. We mostly use this `type` parameter to emit a `DeprecationWarning` from client libraries when a deprecated proto RPC is called. In gax code, we never set the type to anything.

Reply:
going with `UnsupportedParamterWarning` for now, but am open to future adjustment!